app/models/application.rb

```diff
@@ -0,0 +1,16 @@
+class Application < ApplicationRecord
+  has_many :pet_applications
+  has_many :pets, through: :pet_applications
+
+  validates :name, presence: true
+  validates :address, presence: true
+  validates :city, presence: true
+  validates :state, presence: true
+  validates :zip, presence: true
+  validates :description, presence: true
+
+  def add_pet(pet_id)
```

Comment:
I can see what you were trying to do here, add a pet to an application. I think that can be done with the join `pet_application`. So you might have wanted to create a `pet_application_controller` with a `create` action for that association.